workshops/views.py

```diff
@@ -1587,12 +1587,16 @@ class TaskDelete(OnlyForAdminsMixin, PermissionRequiredMixin,
 
 
 class AwardDelete(OnlyForAdminsMixin, PermissionRequiredMixin,
-                  AMYDeleteView):
+                  RedirectSupportMixin, AMYDeleteView):
     model = Award
     permission_required = 'workshops.delete_award'
 
     def get_success_url(self):
-        return reverse('person_edit', args=[self.get_object().person.pk]) + '#awards'
+        next_url = self.request.GET.get('next', None)
+        if next_url:
+            return next_url
+        else:
+            return reverse('badge_details', args=[self.get_object().badge.name])
```

Comment:
```python
class MockAwardDelete(OnlyForAdminsMixin, PermissionRequiredMixin,
                      AMYDeleteView):
    model = Award
    permission_required = 'workshops.delete_award'

    def get_success_url(self):
        return reverse('badge_details', args=[self.get_object().badge.name])


class AwardDelete(RedirectSupportMixin, MockAwardDelete):
    pass
```

This changes the method resolution order, but I think it should work in our case.